src/_pytest/pytester.py

```diff
@@ -524,21 +524,25 @@ class Testdir:
     class TimeoutExpired(Exception):
         pass
 
-    def __init__(self, request, tmpdir_factory):
+    def __init__(
+        self, request, tmpdir_factory, *, monkeypatch: Optional[MonkeyPatch] = None
+    ) -> None:
         self.request = request
-        self._mod_collections = WeakKeyDictionary()
+        self._mod_collections = WeakKeyDictionary()  # type: ignore
         name = request.function.__name__
         self.tmpdir = tmpdir_factory.mktemp(name, numbered=True)
         self.test_tmproot = tmpdir_factory.mktemp("tmp-" + name, numbered=True)
-        self.plugins = []
+        self.plugins = []  # type: ignore
         self._cwd_snapshot = CwdSnapshot()
         self._sys_path_snapshot = SysPathsSnapshot()
         self._sys_modules_snapshot = self.__take_sys_modules_snapshot()
         self.chdir()
         self.request.addfinalizer(self.finalize)
         self._method = self.request.config.getoption("--runpytest")
 
-        mp = self.monkeypatch = MonkeyPatch()
+        if monkeypatch is None:
+            monkeypatch = MonkeyPatch()
+        mp = self.monkeypatch = monkeypatch
```

Comment:
@nicoddemus 
Did not want to make it a required argument, should I?
Should fall back to using `request.getfixturevalue` then though likely (for consistency)?